### PR TITLE
implementation spring data jpa

### PR DESCRIPTION
**Pull request by artemsa223**

### CODE REVIEW

File: src/main/java/com/example/demo/service/BookServiceImpl.java [hunk 1]
```diff
@@ -35,4 +35,32 @@ public BookDto getById(Long id) {
...
+    public BookDto updateById(CreateBookRequestDto requestDto, Long id) {
+        return bookMapper.toDto(bookRepository.findById(id)
+                .map(book -> {
+                    book.setTitle(requestDto.getTitle());
+                    book.setAuthor(requestDto.getAuthor());
+                    book.setIsbn(requestDto.getIsbn());
+                    book.setPrice(requestDto.getPrice());
+                    book.setDescription(requestDto.getDescription());
+                    book.setCoverImage(requestDto.getCoverImage());
+                    return bookRepository.save(book);
```
Comment (Vasul-Pankiv) [inline src/main/java/com/example/demo/service/BookServiceImpl.java]: use mapper to convert requestDto to model

File: src/main/resources/db/changelog/changes/01-create-books-table.yaml [hunk 2]
```diff
@@ -0,0 +1,48 @@
...
+                type: varchar(255)
+            - column:
+                name: cover_image
+                type: varchar(255)
+            - column:
+                name: is_deleted
+                type: boolean
+                defaultValueBoolean: false
+                constraints:
+                  nullable: false
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/changes/01-create-books-table.yaml]: file should end with empty line

File: src/main/resources/db/changelog/db.changelog-master.yaml [hunk 3]
```diff
@@ -0,0 +1,3 @@
+databaseChangeLog:
+  - include:
+      file: db/changelog/changes/01-create-books-table.yaml
```
Comment (Vasul-Pankiv) [inline src/main/resources/db/changelog/db.changelog-master.yaml]: file should end with empty line

**Review by Vasul-Pankiv — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/controller/BookController.java [hunk 4]
```diff
@@ -32,4 +37,20 @@ public BookDto getBookById(@PathVariable Long id) {
     public BookDto createBook(@RequestBody CreateBookRequestDto requestDto) {
         return bookService.createBook(requestDto);
     }
+
+    @GetMapping("/by-title")
+    public List<BookDto> findAllByTitle(@RequestParam String title) {
+        return bookService.findAllByTitle(title);
+    }
```
Comment (Rommelua) [inline src/main/java/com/example/demo/controller/BookController.java]: remove this end point
image

File: src/main/java/com/example/demo/repository/BookRepository.java [hunk 5]
```diff
@@ -2,14 +2,9 @@
...
-public interface BookRepository {
-    Book save(Book book);
-
-    List<Book> findAll();
-
-    Optional<Book> findById(Long id);
+import org.springframework.data.jpa.repository.JpaRepository;
 
+public interface BookRepository extends JpaRepository<Book, Long> {
+    List<Book> findAllByTitleContainsIgnoreCase(String title);
```
Comment (Rommelua) [inline src/main/java/com/example/demo/repository/BookRepository.java]: remove this method

File: src/main/java/com/example/demo/service/BookService.java [hunk 6]
```diff
@@ -11,5 +11,10 @@ public interface BookService {
 
     BookDto createBook(CreateBookRequestDto requestDto);
 
+    List<BookDto> findAllByTitle(String title);
```
Comment (Rommelua) [inline src/main/java/com/example/demo/service/BookService.java]: remove this method

**Review by Rommelua — CHANGES_REQUESTED** [review]

File: src/main/java/com/example/demo/model/Book.java [hunk 7]
```diff
@@ -8,9 +8,13 @@
 import jakarta.persistence.Table;
 import java.math.BigDecimal;
 import lombok.Data;
+import org.hibernate.annotations.SQLDelete;
+import org.hibernate.annotations.Where;
 
 @Data
 @Entity
+@SQLDelete(sql = "UPDATE books SET is_deleted = true WHERE id = ?")
+@Where(clause = "is_deleted=false")
```
Comment (okuzan) [inline src/main/java/com/example/demo/model/Book.java]:
```suggestion
@SQLDelete(sql = "UPDATE books SET is_deleted = TRUE WHERE id = ?")
@Where(clause = "is_deleted = FALSE")
```

**Review by okuzan — APPROVED** [review]
Well done

**Pull request merged**